### PR TITLE
Creating 2 new derived nodes: floor and ceil

### PR DESCRIPTION
**Pull request by rgommezz**
I found an use case where I needed an equivalent `Math.floor` operation, so I thought about creating a new derived node that performs that operation using the existing nodes.

Let me know what you think and if you want me to tweak something :)

Cheers!

### CODE REVIEW

**Comment by osdnk** [comment]
Thank you for this PR, @rgommezz!
I suppose it could be merged, but I suppose in this case it would be also useful to add `ceil`.

File: src/derived/floor.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { add, sub, round } from '../base';
+
+export default function floor(a) {
+  return sub(round(add(a, 0.5)), 1);
```
Comment (osdnk) [inline src/derived/floor.js]: wouldn't it be easier to use `round(sub(a, 0.5))`

**Comment by rgommezz** [comment]
You are welcome!

I've added `ceil` as well.

File: src/derived/ceil.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { add, round, cond, eq } from '../base';
+import floor from './floor';
+
+export default function ceil(a) {
+  return cond(eq(floor(a), a), a, round(add(a, 0.5)));
```
Comment (osdnk) [inline src/derived/ceil.js]: How about `sub(1, round(sub(0.5, a)))`?

File: README.md [hunk 3]
```diff
@@ -302,6 +303,24 @@ round(node)
...
+Returns node that represents the largest integer less than or equal to the input value.
+
+---
+### `ceil`
+
+```js
+ceil(node)
+```
+
+Returns node that represents the largest integer greater than or equal to the input value.
```
[the remark this answers is not in the document]
Comment (rgommezz) [reply README.md]: Will change the API docs to make them more explicit.

File: src/derived/ceil.js (hunk 2 above)
Comment (rgommezz) [reply src/derived/ceil.js]: Yeap, that'll work as well.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (osdnk) [reply README.md]: 👍

**Comment by osdnk** [comment]
Thank you a lot!

**Pull request merged**